Documentation: Run status in kedro-viz (#2436)
kedro-org/kedro-viz: docs/run_status.md

```bash
pip install -r requirements.txt
```

### Run a pipeline

To see your pipeline execution in the Workflow view, execute your pipeline using the standard Kedro command:

```bash
kedro run
```

!!! info
    - The Workflow view is available for full pipeline runs using `kedro run`. Partial pipeline runs (using `--from-nodes`, `--to-nodes`, `--tags`, `--pipeline`, and more) are not supported.
    - SequentialRunner is supported. ParallelRunner and ThreadRunner are not supported.

This will:

- Execute your pipeline
- Automatically collect execution statistics and timing information
- Generate workflow data in the `.viz/kedro_pipeline_events.json` file

Once the run is complete, you can visualise the results:

```bash
kedro viz run
```

Navigate to the **Workflow** view to see your pipeline execution information.

![](./images/workflow_view.png)

## Workflow view scenarios

The following examples show how the Workflow view appears in different pipeline execution scenarios:

### Successful pipeline execution
![Successful pipeline run](./images/run-status-success.gif)

When all nodes execute without errors, you'll see:

> - Green indicators on all nodes showing successful completion
> - A success notification with total execution time
> - Individual node durations displayed over nodes
> - Dataset sizes shown in the node if available 

Reviewer: ```suggestion
- All nodes display green indicators for successful execution
- A success notification shows the total runtime
- Each node displays its execution duration
- Dataset sizes appear on nodes where available 
```

### Node execution failure
![Node execution failure](./images/run-status-node-failure.gif)

When a node fails during execution, you'll see:

> - Red indicators on the failed node
> - An error notification showing which node failed
> - Full error traceback when you click on the failed node

Reviewer: ```suggestion
- The failed node displays a red indicator
- An error notification showing which node failed
- Clicking the node reveals the full traceback error for debugging
```

> - Full error traceback when you click on the failed node

Author: Yes!

- Clear visual display of where the pipeline stopped

### Dataset missing error
![Dataset missing error](./images/run-status-dataset-missing.gif)

When a required dataset is missing, you'll see:

> - The pipeline fails early with specific error information
> - Clear display of which dataset could not be found
> - Helpful error messages to guide you towards resolution

Reviewer: ```suggestion
- The pipeline fails early with clear error message.
- The missing dataset is clearly highlighted in the visualisation
- Helpful error messages to guide you towards resolution
```

> # Run Status tracking in Kedro-Viz

Author: Yes, We need to align on name. @rashidakanchwala @astrojuanlu @iamelijahko

> # Run Status tracking in Kedro-Viz

Reviewer: Visualise and inspect your `kedro run` to debug pipeline execution.

> # Run Status tracking in Kedro-Viz

Reviewer 2: "Workflow view" it is!

> The Run Status feature in Kedro-Viz provides tracking and visualisation of your pipeline execution, giving you detailed insights into your workflow's performance and helping you quickly identify and debug issues.

Reviewer: ```suggestion
From Kedro-Viz 12.0.0 onwards, you can visualise your kedro run to better understand how your pipeline executed. It helps you debug issues by showing which nodes ran successfully, failed, or were skipped — all in the same view.
```

> The Run Status feature in Kedro-Viz provides tracking and visualisation of your pipeline execution, giving you detailed insights into your workflow's performance and helping you quickly identify and debug issues.

Reviewer: ```suggestion
From Kedro-Viz 12.0.0 onwards, the Workflow view lets you visualise your `kedro run` to better understand how your pipeline executed. You can quickly spot which nodes succeeded, failed, or were skipped — making it easier to debug and inspect your workflow.
```

## Overview

Run Status tracks three main aspects of your pipeline execution:

1. **Pipeline execution**: Overall run status, duration, and timing
2. **Node execution**: Individual node performance, success/failure status, and execution time
3. **Dataset operations**: Dataset loading and saving operations with size information

The feature automatically collects execution data during your `kedro run` and presents it in an intuitive visual format within the Kedro-Viz interface.

## Getting started

### Prerequisites

- Kedro-Viz must be installed in your environment
- Your project must be a valid Kedro project

> ### Running a pipeline with Run Status tracking

Reviewer: We should try and use the spaceflights setup section. Please refer to other Kedro-viz doc pages here.

To generate run status data, simply execute your pipeline using the standard Kedro command:

```bash
kedro run

> ```

Author: Done.

This will:
- Execute your pipeline normally
- Automatically collect execution statistics and timing information
- Generate run status data in the `.viz/kedro_pipeline_events.json` file

Once the run is complete, you can visualise the results:

```bash
kedro viz run
```

Navigate to the **Workflow** view to see your run status information.

> 

Author: Done.

> 

Reviewer 2: I think this first screenshot should show the workflow view and not the flowchart view 😄 or, alternative, _just_ the left sidebar to tell users where the icon is

> ## Understanding the Workflow view

Author: Here I am describing the workflow view UI and its components.

> ## Understanding the Workflow view

Reviewer 2: Yeah I think we either call it "Run Status view" or we call the whole feature "Workflow", there's no reason to have 2 separate names I believe?

The Workflow view provides an overview of your pipeline execution with the following components:

### Run Status Notification

After a pipeline run, you'll see a notification banner displaying:

> - **Success notification**: "Run execution completed successfully in [duration]"

Reviewer: I feel we can remove this section. We don't need to describe the UI -- it's quite evident.

- **Success**: Green indicators for successfully executed nodes
- **Failure**: Red indicators for failed nodes
- **Skipped/Not executed**: Gray indicators for nodes that didn't run

## Run Status scenarios

The following examples demonstrate how Run Status appears in different pipeline execution scenarios:

### Successful pipeline execution
![Successful pipeline run](./images/run-status-success.gif)

When all nodes execute successfully, you'll see green indicators and a success notification.

### Node execution failure
![Node execution failure](./images/run-status-node-failure.gif)

When a node fails during execution, you'll see red indicators on the failed node and an error notification.

### Dataset missing error
![Dataset missing error](./images/run-status-dataset-missing.gif)

When a required dataset is missing, the pipeline fails early and shows specific dataset error information.

!!! info
    Currently, Run Status tracking is available for **full pipeline runs only**. Partial pipeline runs (using `--from-nodes`, `--to-nodes`, `--tags`, etc.) are not tracked in the current version.

The following run commands are supported:
- `kedro run` (default pipeline)

> ## API endpoint

Reviewer: We don't need this section in the docs.

> ## API endpoint

Reviewer 2: Yeah I don't think we have API information about anything else... Even if it's just for consistency, +1 to removing it

# Run Status tracking in Kedro-Viz

The Run Status feature in Kedro-Viz provides tracking and visualisation of your pipeline execution, giving you detailed insights into your workflow's performance and helping you quickly identify and debug issues.

## Overview

Run Status tracks three main aspects of your pipeline execution:

1. **Pipeline execution**: Overall run status, duration, and timing
2. **Node execution**: Individual node performance, success/failure status, and execution time

> 3. **Dataset operations**: Dataset loading and saving operations with size information

Reviewer: The Workflow view includes:
- Pipeline execution: Overall run status, start/end time, and duration
- Node execution: Success/failure status, execution time, and performance insights
- Dataset operations: Dataset loads and saves, including file sizes

The feature automatically collects execution data during your `kedro run` and presents it in an intuitive visual format within the Kedro-Viz interface.

## Getting started

### Prerequisites

- Kedro-Viz must be installed in your environment
- Your project must be a valid Kedro project

### Running a pipeline with Run Status tracking

To generate run status data, simply execute your pipeline using the standard Kedro command:

```bash
kedro run
```

!!! warning
    Run Status tracking is currently only available for full pipeline runs using `kedro run`. Partial pipeline runs (using `--from-nodes`, `--to-nodes`, `--tags`, `--pipeline`, etc.) are not supported and will not generate run status data.

This will:

> - Execute your pipeline normally

Reviewer 2: Some formatting issues with this table https://kedro--2436.org.readthedocs.build/projects/kedro-viz/en/2436/run_status/#running-a-pipeline-with-run-status-tracking

- **Failure notification**: "Run execution failed"
- **Completion timestamp**: When the run finished
- **Duration**: Total execution time (for successful runs)

### Visual status indicators

Each node in the workflow view shows visual indicators:

- **Success**: Green indicators for successfully executed nodes
- **Failure**: Red indicators for failed nodes
- **Skipped/Not executed**: Gray indicators for nodes that didn't run

## Run Status scenarios

The following examples demonstrate how Run Status appears in different pipeline execution scenarios:

### Successful pipeline execution
![Successful pipeline run](./images/run-status-success.gif)

When all nodes execute successfully, you'll see green indicators and a success notification.

### Node execution failure
![Node execution failure](./images/run-status-node-failure.gif)

When a node fails during execution, you'll see red indicators on the failed node and an error notification.

### Dataset missing error
![Dataset missing error](./images/run-status-dataset-missing.gif)

When a required dataset is missing, the pipeline fails early and shows specific dataset error information.

!!! warning

>     Currently, Run Status tracking is available for **full pipeline runs only**. Partial pipeline runs (using `--from-nodes`, `--to-nodes`, `--tags`, etc.) are not tracked in the current version.

Reviewer 2: I think this warning is repeated
